apps/server/src/api-integration/integration.controller.ts

```diff
@@ -44,12 +32,21 @@ const actionHandlers: Record<string, ActionHandler> = {
     payload: eventStore.poll(),
   }),
   change: (payload) => {
-    // WS: {type: 'change', payload: { eventId, property, value } }
-    const { eventId, property, value } = payload as ChangeOptions;
-    const { parsedPayload, parsedProperty } = parse(property, value);
-
-    const updatedEvent = updateEvent(eventId, parsedProperty, parsedPayload);
-    return { payload: updatedEvent };
+    assert.isObject(payload);
+    const eventId = Object.keys(payload)[0];
+    assert.isObject(payload[eventId]);
+    const patchEvent: Partial<OntimeEvent> = {};
+    Object.entries(payload[eventId]).forEach(([property, value], _) => {
+      const prop = parse(property, value);
+      if (patchEvent.custom && prop.custom) {
+        Object.assign(patchEvent.custom, prop.custom);
+      } else {
+        Object.assign(patchEvent, prop);
+      }
+    });
+    //TODO: don't know how to await this
+    updateEvent(eventId, patchEvent);
```

Comment:
the function is not async so it cant be awaited. is there an issue here?

Reply:
before i removed it we were trying to return the update event data, but that was always empty as the result was not awaited

if we want that functionality back we need to fix it somehow